packages/website/components/button.js

```diff
@@ -73,16 +78,8 @@ export default function Button({
   const wrapperStyle = { minWidth: small ? '0' : '8rem' }
   const btnStyle = { top: 3, left: 3 }
 
-  let variantClasses = ''
-  switch (variant) {
-    case 'dark':
-      variantClasses = 'bg-black white'
-      break
-
-    case 'light':
-      variantClasses = 'bg-white black'
-      break
-  }
+  // variant can never be null, it has an explicit default of light
+  const variantClasses = buttonVariants[variant]
```

Comment:
technically, this const is unnecessary, but i do think it reads well, just my opinion

Reply:
Some thought and updates:

I got rid of this wrapper class prop, which was not needed for a button to be a button. Now just 'className' is all that's needed to override buttons styles. The button also supports an 'unstyled' prop that was prototyped on the blog PR, and was very useful. This iteration of the button is an improvement on that (@Codigo-Fuentes , if this goes into main before the blog, I will help integrate this change into that branch). I'd like both of you to look over the new button code on this PR if possible, I think you'll like the changes a lot.

As for wrapperClass, I've checked everywhere it was used, and was able to just use 'className'. Way simpler, and more canonical usage of class overrides IMO.

As for the variants, I am applying the variants and we can still type check the expected strings, but the actual implementation is in CSS now, sort of like how semantic UI and several other frameworks handle this. Overrides live in button.css, its open for extension this way. (We can add more colors, sizes and so on if needed).

The only place I saw using the small variant was the hamburger btw, and it wasn't even really 'small' just 'icon sized' so I did not go so far as to rename this but IMO it should not be called 'small' which in most frameworks implies, small...everything. Not in this PR but maybe sometime a button that can take an icon as an argument would be a better abstraction, but this PR kind of does plenty as is so I held back on that.